pkg/apis/messaging/v1beta1/kafka_channel_lifecycle.go

```diff
@@ -122,6 +187,34 @@ func (kcs *KafkaChannelStatus) SetAddress(url *apis.URL) {
 	}
 }
 
+func (kcs *KafkaChannelStatus) MarkConfigTrue() {
+	kcs.GetConditionSet().Manage(kcs).MarkTrue(KafkaChannelConditionConfigReady)
+}
+
+func (kcs *KafkaChannelStatus) MarkConfigFailed(reason, messageFormat string, messageA ...interface{}) {
+	kcs.GetConditionSet().Manage(kcs).MarkFalse(KafkaChannelConditionConfigReady, reason, messageFormat, messageA...)
+}
+
+func (kcs *KafkaChannelStatus) MarkTopicTrue() {
+	kcs.GetConditionSet().Manage(kcs).MarkTrue(KafkaChannelConditionTopicReady)
+}
+
+func (kcs *KafkaChannelStatus) MarkTopicFailed(reason, messageFormat string, messageA ...interface{}) {
+	kcs.GetConditionSet().Manage(kcs).MarkFalse(KafkaChannelConditionTopicReady, reason, messageFormat, messageA...)
+}
+
+func (kcs *KafkaChannelStatus) MarkChannelServiceFailed(reason, messageFormat string, messageA ...interface{}) {
+	kcs.GetConditionSet().Manage(kcs).MarkFalse(KafkaChannelConditionChannelServiceReady, reason, messageFormat, messageA...)
+}
+
+func (kcs *KafkaChannelStatus) MarkChannelServiceTrue() {
+	kcs.GetConditionSet().Manage(kcs).MarkTrue(KafkaChannelConditionChannelServiceReady)
+}
+
+//
+// Consolidated KafkaChannel Condition Markers
+//
```

Comment:
All these implementation-specific `Mark*` methods are confusing, if we add another KafkaChannel implementation do we expect to add more of these?

We should instead move them out of this package and put them as part of the reconciler that uses them. WDYT?

Reply:
Yeah, I'm open to that - should be an easy refactor to separate and move them somewhere like `pkg/channel/consolidated/apis/lifecycle.go` .   Actually, this might cause problems for versioning the api (v1beta1, etc) - so it would have to mirror the apis structure like `pkg/channel/consolidated/apis/messaging/v1beta1/lifecycle.go` ?

Anyone else have an opinion?  If not, I'll refactor as described above ; )

Reply:
The condition set doesn't need versions.

The main reasons are: 
- they're tied to a reconciler architecture and logic, not to the API.
- If the reconciler changes the API that it reconciles to be, let's say, `v1`, the condition set doesn't change automatically.

Reply:
ah, good call - thanks - makes it a little less messy ; )

Reply:
I've refactored them out as suggested.  This means these "custom" lifecycle methods are no longer functions on the KafkaChannelStatus but instead pass-through utilities.  There are pros/cons but overall it seems good ; )